File: template/springboot/TinyPro/src/main/java/com/TinyPro/entity/dto/CreateUserDto.java
package com.TinyPro.entity.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotEmpty;
import jakarta.validation.constraints.NotNull;
import lombok.Data;
import org.springframework.format.annotation.DateTimeFormat;

import java.time.LocalDate;
import java.util.ArrayList;
import java.util.List;

@Data
public class CreateUserDto {
    @NotEmpty(message = "{NOT_EMPTY}")
    private String name;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    @Email
    private String email;
    
    @NotEmpty(message = "{NOT_EMPTY}")
    private String password;
    
    private List<Long> roleIds = new ArrayList<>();
    
    private String department;
    
    private String employeeType;
    @NotEmpty(message = "{NOT_EMPTY}")
    private String probationStart;
    @NotEmpty(message = "{NOT_EMPTY}")
    private String probationEnd;
    private String probationDuration;
    @NotEmpty(message = "{NOT_EMPTY}")
    private String protocolStart;
    @NotEmpty(message = "{NOT_EMPTY}")
    private String protocolEnd;
    
    private String address;
    
    private Integer status;
}

File: template/nestJs/src/user/dto/create-user.dto.ts
import { IsNotEmpty } from 'class-validator';
import { i18nValidationMessage } from 'nestjs-i18n';
import { I18nTranslations } from '../../.generate/i18n.generated';
import { ApiProperty } from '@nestjs/swagger';

export class CreateUserDto {
  @ApiProperty({
    description: '用户名称'
  })
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  name: string;

  @ApiProperty({
    description: '用户邮箱, 登陆时用'
  })
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  email: string;

  @ApiProperty({
    description: '用户密码, 登陆时用'
  })
  @IsNotEmpty({
    message: i18nValidationMessage<I18nTranslations>('validation.NOT_EMPTY'),
  })
  password: string;
  @ApiProperty({
    description: '用户角色'
  })
  roleIds: number[] = [];
  @ApiProperty({})
  department?: string;
  @ApiProperty({})
  employeeType?: string;
  @ApiProperty({})
  probationStart?: string;
  @ApiProperty({})
  probationEnd?: string;
  @ApiProperty({})
  probationDuration?: string;
  @ApiProperty({})
  protocolStart?: string;
  @ApiProperty({})
  protocolEnd?: string;
  @ApiProperty({})
  address?: string;
  @ApiProperty({})
  status?: number;
}
